packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -21,60 +21,25 @@ import '../runner/flutter_command.dart';
 
 /// Map from package name to package version, used to artificially pin a pub
 /// package version in cases when upgrading to the latest breaks Flutter.
+///
+/// These version pins must be pins, not ranges! Allowing these to be ranges
+/// defeats the whole purpose of pinning all our dependencies, which is to
+/// prevent upstream changes from causing our CI to fail randomly in ways
+/// unrelated to the commits. It also, more importantly, risks breaking users
+/// in ways that prevent them from every upgrading Flutter again!
 const Map<String, String> _kManuallyPinnedDependencies = <String, String>{
   // Add pinned packages here. Please leave a comment explaining why.
-  // PACKAGES WITH INCOMPATIBLE LATER VERSIONS
-  // Dart analyzer does not catch renamed or deleted files.
-  // Therefore, we control the version of flutter_gallery_assets so that
-  // existing tests do not fail when the package has a new version.
-  'flutter_gallery_assets': '^1.0.1',
+  'flutter_gallery_assets': '1.0.2', // Tests depend on the exact version.
```

Comment:
Now that we're pinning this exactly, I'm guessing we should let the Flutter Material team know? @pennzht